File: capibara-website/app/dtos/SearchResult.scala
package dtos
import models._
import play.api.libs.json.Json

object SearchResult {
  implicit val searchResultFormatter = Json.format[SearchResult];
}

case class SearchResult(
    macros: Seq[Macro],
    enums: Seq[Enum],
    structs: Seq[Struct],
    typedefs: Seq[Typedef],
    functions: Seq[Function]
)

File: capibara-website/src/dtos/search_result.rs
use rocket::serde::{Deserialize, Serialize};
use  crate::models::_macro::Macro;
use  crate::models::enumeration::Enumeration;
use  crate::models::_struct::Struct;
use  crate::models::typedef::Typedef;
use  crate::models::function::Function;

#[derive(Serialize, Deserialize)]
pub(crate) struct SearchResult {
    pub macros: Vec<Macro>,
    pub enums: Vec<Enumeration>,
    pub structs: Vec<Struct>,
    pub typedefs: Vec<Typedef>,
    pub functions: Vec<Function>
}